ql/src/java/org/apache/hadoop/hive/ql/optimizer/SharedWorkOptimizer.java

```diff
@@ -985,12 +982,45 @@ private static void sharedWorkExtendedOptimization(ParseContext pctx, SharedWork
   /**
    * Obtain the RS input for a mapjoin operator.
    */
-  private static ReduceSinkOperator obtainBroadcastInput(MapJoinOperator mapJoinOp) {
+  private static ReduceSinkOperator obtainFirstBroadcastInput(MapJoinOperator mapJoinOp) {
     return mapJoinOp.getParentOperators().get(0) instanceof ReduceSinkOperator ?
         (ReduceSinkOperator) mapJoinOp.getParentOperators().get(0) :
         (ReduceSinkOperator) mapJoinOp.getParentOperators().get(1);
   }
 
+  private static boolean canShareBroadcastInputs(
```

Comment:
Can we add unit test for this method?

Reply:
Any specific reason why this is a static method? Can we make it non-static?

Reply:
1. Sure, I will add one.
2. I used static method because it does not rely on any of the state of SharedWorkOptimizer instance. I agree that nonstatic method should be preferred to static method and will modify this method and some other static methods such as obtainFirstBroadcastInput() to be nonstatic.